helm_deploy/templates/service.yaml

```diff
@@ -13,3 +13,4 @@ spec:
       name: http
   selector:
     {{- include "laa-submit-crime-forms.selectorLabels" . | nindent 4 }}
+    app: {{ include "laa-submit-crime-forms.fullname" . }}
```

Comment:
I am not sure how this one works - whether it needs the `-web` or not? Prior to this it worked when we had `app:web`.
@jsugarman any ideas?

Reply:
this label renders to "laa-submit-crime-forms-app" already so this would turn to laa-submit-crime-forms-app-web
which feels more confusing to me. But yes, if we add "-web" to the deployment, we'd need to add "-web" to the service label (I tried not matching them and I got errors). I think the reason it worked before was that the service labels were still a subset of the deployment labels

Reply:
for the deployments we never bothered with a `-web` or `-app` in apply. just ` `-worker` suffix in the other deployment.

For the service Apply we have a service-metrics file with  `-metrics` suffix on the label
https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/main/helm_deploy/apply-for-legal-aid/templates/service-metrics.yaml#L17C6-L17C69

plus a service-monitor file that uses matches it

https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/main/helm_deploy/apply-for-legal-aid/templates/service-monitor.yaml

BUT the prometheus rules in CP do not explicitly use these so am not sure how it all wires together and we use a deprecated prometheus gem so not necessarily a good model to follow

Reply:
from the CP docs and Apply's implementation it sounds like we should create a service-metrics.yml separate from service.yml and use the `-metrics` label for that. Then add a service-monitor.yml that matches on the `-metrics` service label. Then we also need a networks-policy.yml to allow prometheus to scrap the `/metrics` endpoint of the app.

Reply:
yeah like Jo said in Slack - Apply is using metrics in a different way. They have Prometheus Exporter and [deploy a separate container with prometheus exporter running](https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/main/helm_deploy/apply-for-legal-aid/templates/deployment_metrics.yaml). Whereas we use promethues client and add that client as middleware to our existing application so the Service of the application and metrics is shared. 

I think it depends on whether we see benefits in doing things this way or that way

Reply:
Discussed and resolved